### PR TITLE
Support rendering proxy objects.

### PR DESCRIPTION
**Pull request by loganfsmyth**
Refs https://github.com/RecordReplay/backend/issues/3516

### CODE REVIEW

File: src/protocol/thread/value.ts [hunk 1]
```diff
@@ -152,6 +152,11 @@ export class ValueFront {
     return (this.hasPreview() && this._object!.preview!.promiseState) || null;
   }
 
+  previewProxyState() {
+    // Older recordings did not set proxyState so this could be null.
+    return (this.hasPreview() && this._object!.preview!.proxyState) || null;
```
Comment (jaril) [inline src/protocol/thread/value.ts:157]: Small nit, but what do you think about doing: `return this._object?.preview?.proxyState || null`.

Comment (jaril) [reply src/protocol/thread/value.ts:157]: Meh could go either way, tbh this is more readable!

**Review by jaril — APPROVED** [review]
LGTM!

File: src/protocol/thread/value.ts (hunk 1 above)
Comment (loganfsmyth) [reply src/protocol/thread/value.ts:157]: This was essentially a copy-paste from the function above this, so probably in favor of leaving it.

**Pull request merged**